templates/vue/src/components/TapestryMain/RootNodeButton/index.vue

```diff
@@ -148,15 +148,33 @@ export default {
             this.changes.permissions.add(key)
           }
         }
-        if (node.permissionsOrder) {
-          node.permissionsOrder = node.permissionsOrder.filter(role =>
-            wp_roles.has(role)
-          )
-        }
         if (this.changes.permissions.size > 0) {
           this.changes.noChange = false
         }
       }
+      if (!data.settings) {
+        return
+      }
+      let originalDefaultPerms = data.settings.defaultPermissions
+      data.settings.defaultPermissions = Object.keys(
+        data.settings.defaultPermissions
+      )
+        .filter(key => wp_roles.has(key))
+        .reduce((obj, key) => {
+          return {
+            ...obj,
+            [key]: data.settings.defaultPermissions[key],
+          }
+        }, {})
+      const keys = Object.keys(data.settings.defaultPermissions)
+      for (let key in originalDefaultPerms) {
+        if (!keys.includes(key)) {
+          this.changes.permissions.add(key)
+        }
+      }
+      if (this.changes.permissions.size > 0) {
+        this.changes.noChange = false
+      }
```

Comment:
I find this function a bit hard to read. Would you mind adding some spacing and comments to clarify to the reader what's going on?